notebooks/233-blip-visual-language-processing/233-blip-optimize.ipynb

```diff
@@ -0,0 +1,652 @@
wt//HjBtMZGY/SOxVu+Xg9+er7z623Hew/WokzGWdGZ1A51iyEIL6Q3rHK5/HbB5CIag/WoPK8H//nnFwYDFsXDNPO3Pzzw+enCnhK3pFjzWzCfQLz8IOu81jkd9+uN7XYXWF2KHOtG2PeevNrTH43Y5E13SVnrRPDcJwc5SgAYCMa/tV4sfZz5H//4wDyO/OdfVo5i2Pe/kAbBGt0JcKpjIOWForsIjlzJW+hWw4ZO9X2H5JwRxb7SeO8ld8CLXaqW1hPOhEktCqBGa7/FEJueTCW+6/6fc5JdZtMyQgUpDgDdRPCG0u+FBsjYVr3tkI1Yt5TuNDSjmZa5dzoyQnfTiFWNY914/faM7pTC4wjdR79hnWTY+2FgGNx716StJh2BkuQXLl+iBztg/SAshhhkH5qzFDWdrhWOREid1uUt2llqCMQjMJ5OcmEnEbqVkDEuM1gnVX7XdvhRkjRp4sgQhbgo6XVPZnxT2Sol8bO1VY7rTtgP0XTQcOMgFxqIHS4X8nFQk3iB1/VOThk3jTTdR+tJKluJgy54k9nWrpFQAqFJIRKPg5wCpu+Ax2XhLdBm3wN7KqAMfhrl50BLzgagVCOnyHyaSSlLNvztzrd1659bUULCVRjGAXpxZhoMY30zlICCaRYtSKhRBJhdWKm0ktFhlcx2g+yHVXchCN7XYI2VQtaKSCn3Xf8bHMgYzdC8xNnm3He3mtpXYvM8c1vv4lRpjZJkpaOsuDxqSjQaaRMcd6kycsy1MkwatR/MpzMxBhyiiTgiPJgzVkvoiza2M0AqWzigFtZdmO7TNDEOI6ff/Y5WRRMQUmbft+6yUIzjyDzNvJFK5T2Rz6ZqoZ1H0scHgXg59/4dFGSUnftlrBA3glgQG9YO4mZSjSNKNLM2kpOgtTQM2hhCCtQmUdnGGmKKkvyXqwR7GVnzSVKeKONjisIJyPJ/Py8TWmuJ6u3n5ltIkdGdbKdlRReyJPrtewClOEqFkmXNoRX7sRNCkt93SkKdTKJyj/tGyTO263isd6Qti+C6ZmqRULRSG2kXD/yxHRQv9mddKqAljVMY3EzLQgOO7UCVhm29Cz4CWsuacj7NaCudvkJSK1VQhLyBeetGDcpWYjqIx96jibd+qWdQSFaK0iIc10aGrmimy4XxEDJuONL78/SWtmo6Nr+k8r4CHo3uHJGAMwKze153nr9+pTWZLlnrccNISZlUkoC7hkEos53YahU4o6kKYsoE6zjGkfu/fMPSSEdido2HZeSHh4kPpxldDj6eR45UQFvOduTrrbD3SYt1/02K8H8Djyql4LsuLmehldZaOXqTXKrY7g2VP91mPi6eaXQUr6HJlE0jk2mAEBLrurHdV15fXruOyPRQMJkelFpJHbVejRQTb/+86XneKLByFL9puRTTYPnj9498f5nYsuPHXyN/uitCZ+38m+73///Kgt/+iUEsYdbIKE73jo++cqDV97GqjMjLe/iJ9w4qlJrY+wEAgl5G8W4F096Kcl7JJZ/eqiTXHQW1p2i1RtiEh1AbqKb6ftCgmuwYVWcjUButW8m0Nu+2oXGZsc4zTJP8MnMWZffbAdaTBrd9Ja472nsGpd+VpcoY1mPHJtMV2UI69MPAMk0i6BkHVq0I+06tjeFyYRwm2enth6wxSkbyp9X75yutoIumpIrSnfS2i2fcOEeJhaobfpzRo2KcJ/w0yuVcMq2L5QTl2aRQQrFvB0aDdSLWGoYJN4zCjKhNaGZdnS0IEiXpcFXWF41GRUiZuslU4q1rVsYwXWaMEjpirQ1lLa5JIVaz/DsxirJv6A6s0tZihwHdFMYPOKN5enwgJQGjvKnTU85o7WgtE6O8xNaarjmQFzKWQm7CXpCoY4lPPqEY5plawWklu+VcMCgG71m3/V2Ed6RCbq2r2BVOa8FXKyXY5NZ5BVVGiFLkZqwy3cMs+3DrhaOvtX7vuE0RyFZFvus9ymRpHgce7AVlTJ+YiHDT+AHnB0k33DdMLWTn5HejTF/RyK5/9BNUEfymnDoSfWcax56BMUp4VUPWVjmJJWwYKD1e2eieaWE00zTxdD6z7TvKiEvAOUcIByWBQbGuK5fLSXb+CgbfKFnIn9VKMFtMGRCh4tGhWFpLimHJqU8WlawucuIIAWutWAxLQTUYnJcsDwXDPMoOP0uKaaNRaiX2QDbTRcTivlGY0mhH4Jdvzzw+XDBoLstETF7OFyUrsOM4MJ29QJOC0Hn/Tn01znHbRWznvUdbT96CrK+UPC0ogSFp6ziOhEOhDd2mJgAlpQy1Sr6MtlZC0rTDTxCPRCniBtLW02ISMWLOYKSZyalPL0Kk5Fe0tfhp7hA70XEordi3jbXKsxmP0CdZMn0Zx4HttvH69RfRVoQqDhs3dr2V68WZoPNphUrAzwXtvEz8jAUlEzLVNSS5FPbuqjg/XASh//oiz64xHNcbgzf480nMGushrgDnmLo48whBbILrjdUPWO+Z50XcUjFK0aMV3lu++/iBoDT7nji2DUrin192Zg2mNrypPFrhzVQz8jVUjtK4HeKSEVsioGWSt6+rnAVWIt21ku+ZWqV5pWKapVXhzyTgl9vG1+tdYFjOUpGGWSlQ5Y3uKc9m6vqm2hrpiEAVrVupHbUt09TURNwIvGc7pJRoUVxob5wQY4wUUH7keS287Bu5QtNdkxcTR8/k+D+9QEhFXj5vuve7ixDfFP19SYDCUKlo41BKi2CvRJTud5XSKKPed9XWewkuaaV/sd0poIyMxJV6P6yhBz1ZS71c5IDsVVI4RIDUVBc4GsMwCfe9ZAl6aQhJzg1eglP6BVRyYQ+BGBPQx8SNLsjJ6HHAWgnUkL4aEfX1w6/GLoqrDRclaU+/CbCcw/cdhjXmXamsrJbRUkyMy9w7T7nUaOCnSTDDVhj1/ukRo4TQNZ0XnHcSvoSSSY2G2hRFW1qfCriegliLHADDKBnpukdVS6ys/E60MRzHQbMG1Sw1R1rh/bBMSXbzFd4T5
```

Comment:
steps with calculate_inference_time() work really long. We should at least add a note or maybe reduce the dataset somehow

Reply:
I reduced dataset to 100 examples (x3). Locally `calculate_inference_time` works for about ~30s.